### DISCUSSION TITLE
Which tools are used to build Instrumenta?

### DISCUSSION THREAD

**Discussion by EBITDAFF** [iappyx/Instrumenta#34] [General]
Hi iappyx,

I'm thinking about creating a similar add-in on Excel, and wanted to know if you have any best practices to share please?
For example, do you recommend any specific Visual Studio Code extension to edit the files?

Thanks

**Comment by iappyx** [comment]
Not sure if they are really best practices. I'm just using the VBA IDE of PowerPoint. I code and test the file as PowerPoint file with enabled macros. To customize the Ribbon I am using [https://github.com/fernandreu/office-ribbonx-editor](url).

When all works ok (both on Windows and Mac), I save the file in a PowerPoint add-in format. Very simple, but for me this works well.

You can also go a bit more professional and go beyond VBA (see [https://learn.microsoft.com/en-us/office/dev/add-ins/develop/develop-overview](url)) but I have not yet taken that step.

**Reply by EBITDAFF** [reply]
Many thanks iappyx!
